homeassistant/components/analytics/analytics.py

```diff
@@ -420,6 +537,25 @@ async def async_devices_payload(hass: HomeAssistant) -> dict:
             integration_domain, {"devices": [], "entities": []}
         )
 
+        modifiable_device_info = DeviceAnalytics(
+            sw_version=device_entry.sw_version,
+        )
+
+        if (
+            device_modifier := await _async_get_device_modifier(
+                hass, integration_domain
+            )
+        ) is not None:
+            try:
+                await device_modifier(hass, device_entry.id, modifiable_device_info)
```

Comment:
We should not do this per device, that is very inefficient. Some integrations are more efficient gathering all the info for their devices at once. 

I would suggest we ask for overrides for a specific set of device IDs and get a dictionary back mapping device ID to filter keys/extras.